Source/WebCore/css/MediaQueryEvaluator.cpp

```diff
@@ -481,6 +481,33 @@ static bool dynamicRangeEvaluate(CSSValue* value, const CSSToLengthConversionDat
     }
 }
 
+static bool scanEvaluate(CSSValue* value, const CSSToLengthConversionData&, Frame& frame, MediaFeaturePrefix)
+{
+    auto view = frame.view();
+    if (!view)
+        return false;
+
+    /* With Media Queries Level 4, the "tv" media type is deprecated
+     * and the "scan" feature applies to all media types.
+     * We are currently supporting Media Queries Level 3,
+     * thus the check against "tv".
+     */
+    auto mediaType = frame.view()->mediaType();
+    static constexpr auto tv = ASCIILiteral::fromLiteralUnsafe("tv");
+    if (!equalIgnoringASCIICase(mediaType, tv))
+        return false;
+
+    if (!value)
+        return false;
+
+    if (!is<CSSPrimitiveValue>(*value))
```

Comment:
... and drop the `*` here so that it does the null check for you.

Reply:
But really, I would just write this:
```
auto* primiveValue = dynamicDowncast<CSSPrimitiveValue>(value);
if (!primiveValue)
    return false;
```

No need for earlier `value` null check or for the downcast below.